readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/HdfsExporter.java

```diff
@@ -88,15 +90,23 @@ public static void main(String[] args) throws IOException {
         final String baseTemporaryHdfsDir = config.getHdfs().getBaseTemporaryDir();
         final Path finalHdfsDir = new Path(config.getHdfs().getFinalDir());
 
-
-        FileSystem fs = null;
         try {
             fs = finalHdfsDir.getFileSystem(HDFS_CONF);
         } catch (IOException e) {
             LOGGER.error("Could not initialize HDFS", e);
             exit(1);
         }
 
+        // Add Shutdown Hook to close FS and delete temp folders
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                fs.close();
+            } catch (IOException e) {
+                LOGGER.error("Could not close FS", e);
+                exit(1);
```

Comment:
why do you need this shutdown hook? it will be closed anyway by the shutdown of the JVM or is it required by the deleteOnExit?

Reply:
How could it be closed during the shutdown of the JVM?
To process deletion of deleteOnExit path fs object should call the processDeleteOnExit method which is called on the close method.
This is why I'm using this shutdown hook. Also it ensure that the close "action" will be closed in almost all cases, only JVM crash and SIGKILL will leads to this action not being called.

Reply:
there is already a shutdown hook for filesystems in FileSystem.Cache.getInternal

Reply:
Good catch